doc-content/drools-docs/src/main/asciidoc/DecisionEngine/cep-negative-patterns.adoc

```diff
@@ -0,0 +1,60 @@
+[id='cep-negative-patterns_{context}']
+
+= Negative patterns in {DECISION_ENGINE} stream mode
+
+A negative pattern is a pattern for conditions that are not met. For example, the following DRL rule activates a fire alarm if a fire is detected and the sprinkler is not activated:
+
+.Fire alarm rule with a negative pattern
+[source]
+----
+rule "Sound the alarm"
+when
+  $f : FireDetected()
+  not(SprinklerActivated())
+then
+  // Sound the alarm.
+end
+----
+
+In cloud mode, the {DECISION_ENGINE} assumes all facts (regular facts and events) are known in advance and evaluates negative patterns immediately. In stream mode, the {DECISION_ENGINE} can support temporal constraints on facts to wait for a set time before activating a rule.
+
+The same example rule in stream mode activates the fire alarm as usual, but applies a 10-second delay. The `duration()` tag in the rule is optional.
+
+.Fire alarm rule with a negative pattern and time delay (stream mode only)
+[source]
+----
+rule "Sound the alarm"
+when
+  $f : FireDetected()
+  not(SprinklerActivated(this after[0s,10s] $f))
+then
+  // Sound the alarm.
+end
+----
+
+.Fire alarm rule with a negative pattern, duration parameter, and time delay (stream mode only)
+[source]
+----
+rule "Sound the alarm"
+  duration(10s)
```

Comment:
In original docs the duration was used to create an analogy, to let the reader better understand the behaviour of temporal negative patterns. Either you keep the same analogy or remove this. Saying that you could optionally add a duration is not the same thing and I actually find it more confusing.

Reply:
Done. Removed the second example with the duration parameter, and the bit about it being optional.